### ISSUE TITLE
RFE: Server/Client version

### ISSUE THREAD

**Issue by MrMEEE** [stuartthomas25/JParty#28]
Hi guys

Thanks for a GREAT job on JParty..

For me (and probably others), one thing that would really improve this project, is the ability to split into a server and a client/admin component.

I think the following steps are needed:
- Split the code into backend and frontend components
- Depending on the current design, some kind of API may be needed?
- Authentication from the frontend (admin) towards the backend

This would provide the following:
- Ability to run without a shared local network, therefore also using this through google meet, teams or similar
- In time, being able to use two separate devices for the game screen and the admin screen.

Is this something that you have thought about?

I would love to join on getting this done..

**Comment by rosslandgraf** [comment]
@MrMEEE Do you mean something like hosting the game on a server so that people can connect in without being in person? If so this is something we have been talking about implementing in our fork at some point. [https://github.com/gdsimco/JParty-with-buzzers](Our Fork)

**Comment by MrMEEE** [comment]
Yeah, basically splitting it into the following components:- server- client (buzzers)- admin (game master)- game boardAnd making the components all talk into the server (which can be hosted online)Sent from Android device26. dec. 2023 07.40 skrev Ross Landgraf ***@***.***>:
@MrMEEE Do you mean something like hosting the game on a server so that people can connect in without being in person? If so this is something we have been talking about implementing in our fork at some point. [https://github.com/gdsimco/JParty-with-buzzers](Our Fork)

**Comment by stuartthomas25** [comment]
I think would entail a fairly large redesign of the program. One option for non-local play is using a VPN. I've verified that the buzzer and host can be on different networks and still communicate via a personal VPN.
